modules/s4/ansible.tf

```diff
@@ -0,0 +1,219 @@
+  image = "projects/rhel-cloud/global/images/rhel-9-v20230615"
+  lifecycle {
+    ignore_changes = [snapshot, image]
+  }
+  name    = "${var.deployment_name}-ansible-runner"
+  project = data.google_project.sap-project.project_id
+  size    = 50
+  timeouts {
+    create = "1h"
+    delete = "1h"
+    update = "1h"
+  }
+  type = "pd-ssd"
+  zone = var.zone1_name
+}
+
+resource "google_compute_instance" "sapdansible11" {
+  allow_stopping_for_update = var.allow_stopping_for_update
+  boot_disk {
+    auto_delete = false
+    device_name = "persistent-disk-0"
+    source      = google_compute_disk.sapdansible11.self_link
+  }
+  depends_on = [
+    google_storage_bucket_object.ansible_inventory,
+    google_compute_instance.sapdapp11,
+    google_compute_instance.sapddb11,
+    google_compute_instance.sapdascs11
+  ]
+  labels = {
+    active_region  = true
+    component      = "ansible"
+    component_type = "generic"
+    environment    = var.deployment_name
+    service_group  = "ansible_runner"
+  }
+  lifecycle {
+    ignore_changes = [
+      min_cpu_platform,
+      network_interface[0].alias_ip_range,
+      metadata["ssh-keys"]
+    ]
+  }
+  machine_type = "n1-standard-16"
+  metadata = {
+    active_region             = true
+    configuration_bucket_name = data.google_storage_bucket.configuration.name
+    dns_zone_name             = data.google_dns_managed_zone.sap_zone.name
+    is_test                   = var.is_test
+    media_bucket_name         = var.media_bucket_name
+    ssh-keys                  = ""
+    startup-script            = "gsutil cp ${var.primary_startup_url} ./local_startup.sh; bash local_startup.sh ${var.package_location} ${var.deployment_name}"
+    template_name             = "s4"
+    ansible_sa_email          = data.google_service_account.service_account_ansible.email
+    app_sa_email              = data.google_service_account.service_account_app.email
+    ascs_sa_email             = data.google_service_account.service_account_ascs.email
+    db_sa_email               = data.google_service_account.service_account_db.email
+  }
+  name = "${var.deployment_name}-ansible-runner"
+  network_interface {
+    dynamic "access_config" {
+      content {
+      }
+      for_each = var.public_ansible_runner_ip ? [1] : []
+    }
+    network    = data.google_compute_network.sap-vpc.self_link
+    network_ip = google_compute_address.sapdansible11-1.address
+    subnetwork = data.google_compute_subnetwork.sap-subnet-ansible-1.self_link
+  }
+  project = data.google_project.sap-project.project_id
+  scheduling {
+    automatic_restart   = true
+    on_host_maintenance = "MIGRATE"
+    preemptible         = false
+  }
+  service_account {
+    email  = data.google_service_account.service_account_ansible.email
+    scopes = ["https://www.googleapis.com/auth/cloud-platform"]
+  }
+  tags = ["${var.deployment_name}-s4-comms"]
+  zone = var.zone1_name
+}
+
+resource "google_project_iam_member" "ansible_sa_role_1" {
```

Comment:
At the moment making this change in TF would be difficult due to the build processes we use. So this could be useful in the future but for now can't be done.